model/move.ts

```diff
@@ -0,0 +1,24 @@
+import { match, P } from "ts-pattern";
+import { z } from "zod";
+
+export enum Moves {
+  Rock = "Rock",
+  Paper = "Paper",
+  Scissors = "Scissors",
+}
+
+const MoveEnum = z.nativeEnum(Moves);
+export type MoveEnum = z.infer<typeof MoveEnum>;
```

Comment:
In this specific example, inferring the type `MoveEnum` is useless because it will just infer the same type as `Moves`.

In general, when using zod we want to:

- define the schema using zod utilities
- infer the type using `z.infer`